### DISCUSSION TITLE
Where to setup shfmt's option in nvim?

### DISCUSSION THREAD

**Discussion by dlintw** [LazyVim/LazyVim#1476] [General]
Don't know how to setup "shfmt -i 2 -ci -bn" as format shell script option.

**Comment by dlintw** [comment]
After search and read https://github.com/LazyVim/LazyVim/discussions/1081

I try to add 2 lines in config/lazy.lua as

```
local null_ls = require("null-ls")
null_ls.builtins.formatting.shfmt.with({ extra_args = { "-i", "2", "-ci", "-bn" } })
```

And it does not work.

**Comment by dlintw** [comment]
After read https://github.com/LazyVim/LazyVim/discussions/611, I figure out the workable method.

We can use one of two methods to setup:
1.  add a file  ~/.config/nvim/lua/plugins/nullls.lua
2. modify the ~/.config/nvim/lua/config/lazy.lua

For shorter 2 or 3 lines settings,  I prefer use method 2. for longer lines, maybe method 1 is better.  However, I keep versions of my settings in 1~3 setting files, So, I'll try to use method 2.   Or, maybe it is possible put in ~/config/nvim/lua/config/options.lua,  That's better place for me to put the option settings.

Method 1:
```
return {
    {
      "jose-elias-alvarez/null-ls.nvim",
      opts = function(_, opts)
        local nls = require("null-ls")
        opts.sources = vim.list_extend(opts.sources, {
          nls.builtins.formatting.shfmt.with({
            extra_args = { "-i", "2", "-ci", "-bn" },
          }),
        })
      end,
    },
}
```
Method 2:
```
require("lazy").setup({
  spec = {
    -- add LazyVim and import its plugins
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    # -- omit other plugins settings
    {
      "jose-elias-alvarez/null-ls.nvim",
      opts = function(_, opts)
        local nls = require("null-ls")
        opts.sources = vim.list_extend(opts.sources, {
          nls.builtins.formatting.shfmt.with({
            extra_args = { "-i", "2", "-ci", "-bn" },
          }),
        })
      end,
    },
    { import = "plugins" },
  },
  -- omit others ...
})
```

**Reply by dpetka2001** [reply]
Glad you were able to figure it out. I personally prefer always method 1 as it makes more sense to me as how to structure my configuration. In `/lua/config/lazy.lua` I only enable the `extras` Lazyvim provides that I'd like. Everything else goes to `/lua/plugins/some_file.lua`. 
The file for options `/lua/config/options.lua` is only for setting Neovim options such as the ones you can see [here](https://www.lazyvim.org/configuration/general#options) if you press on the `Default Options` tab. I don't believe it's able to parse plugin specs in `/lua/config/options.lua`.
